data_labeling_examples/src/main/java/com/oracle/datalabelingservicesamples/scripts/CustomBulkLabelingScript.java

```diff
@@ -0,0 +1,177 @@
+import com.oracle.bmc.datalabelingservicedataplane.model.Entity;
+import com.oracle.bmc.datalabelingservicedataplane.model.GenericEntity;
+import com.oracle.bmc.datalabelingservicedataplane.model.Label;
+import com.oracle.bmc.datalabelingservicedataplane.model.LabelName;
+import com.oracle.bmc.datalabelingservicedataplane.model.Record.LifecycleState;
+import com.oracle.bmc.datalabelingservicedataplane.model.RecordSummary;
+import com.oracle.bmc.datalabelingservicedataplane.requests.CreateAnnotationRequest;
+import com.oracle.bmc.datalabelingservicedataplane.requests.GetDatasetRequest;
+import com.oracle.bmc.datalabelingservicedataplane.requests.ListRecordsRequest;
+import com.oracle.bmc.datalabelingservicedataplane.responses.GetDatasetResponse;
+import com.oracle.bmc.datalabelingservicedataplane.responses.ListRecordsResponse;
+import com.oracle.datalabelingservicesamples.requests.Config;
+
+import lombok.extern.slf4j.Slf4j;
+
+/*
+ * 
+ * This Script takes following input for Bulk Labeling:
+ * 
+ * DATASET_ID : the id of the dataset that you want to bulk label
+ * CUSTOM_LABELS : JSON input specifying the labels to be applied for a object storage path.
+ * 					Ex: { "dog/": ["dog"], "cat/": ["cat"] }
+ * LABELING_ALGORITHM : The pattern matching algorithm that will determine the label of any record. 
+ * 						Currently following algorithms are supported: CUSTOM_LABELS_MATCH
+ * 
+ * 
+ * Following code constraints are added:
+ * 	1. At max 3 distinct path is accepted 
+ *	2. No nested path is allowed. Only top-level path is considered.
+ *	 In case the user wants nested folder support, he/she can place the nested folder at top level to use this API.
+ * 	3. The API only annotates unlabeled records. Labels that are already annotated will be skipped.
+ * 
+ */
+@Slf4j
+public class CustomBulkLabelingScript {
+
+	static ExecutorService executorService;
+	static Dataset dataset;
+	static List<String> successRecordIds = Collections.synchronizedList(new ArrayList<String>());
+	static List<String> failedRecordIds = Collections.synchronizedList(new ArrayList<String>());
+
+	static {
+		executorService = Executors.newFixedThreadPool(Config.INSTANCE.getThreadCount());
+	}
+
+	public static void main(String[] args) throws InterruptedException, ExecutionException {
+		long startTime = System.nanoTime();
+		String datasetId = Config.INSTANCE.getDatasetId();
+		Map<String, List<String>> bulkLabelingRequest = Config.INSTANCE.getCustomLabels();
+		validateRequest(datasetId, bulkLabelingRequest);
+		log.info("Starting Bulk Labeling for dataset: {}", dataset.getDisplayName());
+		Boolean isUnlabeledRecordAvailable = true;
+		String page = null;
+
+		while (isUnlabeledRecordAvailable) {
+			ListRecordsRequest listRecordsRequest = ListRecordsRequest.builder().datasetId(datasetId)
+					.compartmentId(dataset.getCompartmentId()).lifecycleState(LifecycleState.Active).page(page)
+					.limit(1000).build();
```

Comment:
Shall we not mention this limit of 1000 in README.md ? Also, we should move this value to a constants file.

Reply:
I did not mention 1000 in README.md as the user need not change that value. It's more of an implementation part. I have moved the value to a constant file.